filemanager/gcsmanager.go

```diff
@@ -28,18 +30,32 @@ type GCSConfig struct {
 	ForcePathStyle *bool
 	DisableSSL     *bool
 	JSONReads      bool
+
+	doNotOverWrite bool
+}
+
+type Opt func(*GCSConfig)
+
+func WithNoOverwriteGCS(doNotOverride bool) func(*GCSConfig) {
+	return func(c *GCSConfig) {
+		c.doNotOverWrite = doNotOverride
+	}
 }
 
 // NewGCSManager creates a new file manager for Google Cloud Storage
 func NewGCSManager(
-	config map[string]interface{}, log logger.Logger, defaultTimeout func() time.Duration,
+	config map[string]interface{}, log logger.Logger, defaultTimeout func() time.Duration, opts ...Opt,
 ) (*gcsManager, error) {
+	gcsConfig := gcsConfig(config)
+	for _, opt := range opts {
+		opt(gcsConfig)
+	}
```

Comment:
We can probably rename it as conf since it shadows the declaration of 'gcsConfig' func.
```suggestion
	conf := gcsConfig(config)
	for _, opt := range opts {
		opt(conf)
	}
```